Science/CanadianElectricityGeneration/canadian-electricity-generation.ipynb

```diff
@@ -224,6 +224,37 @@
     "px.sunburst(df[df['Type']=='Territory'], path=['Area','Renewable','Source'], values='Proportion', title='Electricity Generation in the Territories')"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Alternate Data Sources\n",
+    "\n",
+    "We can also obtain data from tables on a Wikipedia article. The data are likely more current, but may contain errrors or stations that are not yet operational.\n",
+    "\n",
+    "As an example, let's get the data for [electrical generating stations in Alberta](https://en.wikipedia.org/wiki/List_of_generating_stations_in_Alberta). The article format or ordering may have changed and the code may need to be updated to match, it would be a good idea to compare the article to the categories list in the code."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "df = pd.read_html('https://en.wikipedia.org/wiki/List_of_generating_stations_in_Alberta')\n",
+    "categories = ['coal','natural gas','dual fuel','biomass','geothermal','hydroelectric','wind','solar']\n",
+    "values = []\n",
+    "for i, c in enumerate(categories):\n",
+    "  total = int(pd.to_numeric(df[i+1]['Capacity (MW)'], errors='coerce').sum())\n",
+    "  values.append(total)\n",
+    "print(categories)\n",
+    "print(values)\n",
+    "\n",
+    "import plotly.express as px\n",
```

Comment:
```import plotly.express as px``` statement here is unnecessary with import in first code cell

Reply:
You're right, the duplicate `import` statements are not necessary. I sometimes include them in case a cell is being run on its own, rather than the notebook being run from top to bottom. I'll remove the duplicates in this case though.